code/API_definitions/location-subscriptions.yaml

```diff
@@ -0,0 +1,591 @@
+      description: The request for creating a Location Event Subscription
+      type: object
+      properties:
+        webhook:
+          $ref: '#/components/schemas/Webhook'
+        subscriptionDetail:
+          $ref: '#/components/schemas/SubscriptionDetail'
+        subscriptionExpireTime:
+          type: string
+          format: date-time
+          example: 2023-01-17T13:18:23.682Z
+          description: The time when the location-tracking has to be terminated (e.g. in 14 days from now on). Provided in date-time format.
+      required:
+        - webhook
+        
+    SubscriptionInfo:
+      description: Represents a area-location event subscription.
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/CreateEventSubscription'
+        - type: object
+          properties:
+            eventSubscriptionId:
+              $ref: '#/components/schemas/EventSubscriptionId'
+            startsAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking has to be started (e.g. in 7 days from now on). Provided in date-time format.
+            expiresAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking has to be expired (e.g. in 7 days from now on). Provided in date-time format.
+          required:
+            - eventSubscriptionId
+            - webhook
+  
+    EventTime:
+      format: date-time
+      type: string
+      description: The time when the event was notified.
+    
+    EventNotifications:
+      description: The event notification callback.
+      type: object
+      required:
+        - event
+      properties:
+        eventSubscriptionId:
+          $ref: '#/components/schemas/EventSubscriptionId'
+        event:
+          $ref: '#/components/schemas/Event'
+    
+    Event:
+      description: The event being notified
+      anyOf:
+        - $ref: '#/components/schemas/AreaEnteredEvent'
+        - $ref: '#/components/schemas/AreaLeftEvent'
+      discriminator:
+        propertyName: eventType
+        mapping: 
+          AREA_ENTERED: '#/components/schemas/AreaEnteredEvent'
+          AREA_LEFT: '#/components/schemas/AreaLeftEvent'
+    
+    EventId:
+      type: string
+      description: The subscription event identifier.
+    
+    EventBase:
+      type: object
+      required:
+        - eventType
+        - eventTime
+      properties:
+        eventId:
+          $ref: '#/components/schemas/EventId'
+        eventType:
+          $ref: "#/components/schemas/EventType"
+        eventTime:
+          $ref: '#/components/schemas/EventTime'
+    
+    AreaEnteredEvent:
+      allOf:
+        - $ref: "#/components/schemas/EventBase"
+        - type: object
+          properties:
+            eventDetail:
+              type: object
```

Comment:
We have to provide the details of the event for this specific event

Reply:
Please check the last commit.